learningresources/models.py

```diff
@@ -79,14 +82,38 @@ def has_resources(self):
 
     @transaction.atomic
     def save(self, *args, **kwargs):
-        """Handle slugs."""
+        """Handle slugs and groups"""
+        is_create = False
+        is_update = False
         if self.id is None or self.name != get_object_or_404(
                 Repository, id=self.id).name:
+            # if it is an update of the repository, need the old slug
+            if self.id is not None and \
+                    self.name != get_object_or_404(Repository,
```

Comment:
nit: I actually though \ ticks for line continuation was a pep8 violation: I would prefer
```python
if (
            self.id is not None and
            self.name != get_object_or_404(
                Repository,
                id=self.id
            ).name
):
```